xtask/src/themelint.rs

```diff
@@ -0,0 +1,172 @@
+    fn has_fg(fg: String) -> Rule {
+        Rule {
+            fg: Some(fg),
+            bg: None,
+            check_both: true,
+        }
+    }
+    fn has_either(item: String) -> Rule {
+        Rule {
+            fg: Some(item.clone()),
+            bg: Some(item),
+            check_both: false,
+        }
+    }
+    fn has_both(item: String) -> Rule {
+        Rule {
+            fg: Some(item.clone()),
+            bg: Some(item),
+            check_both: true,
+        }
+    }
+    fn validate(&self, theme: &Theme, messages: &mut Vec<String>) {
+        let mut found_fg = true;
+        let mut found_bg = true;
+        let mut fg_name = "";
+        let mut bg_name = "";
+        if let Some(fg) = &self.fg {
+            fg_name = fg;
+            if theme.get(fg).fg.is_none() && theme.get(fg).add_modifier == Modifier::empty() {
+                found_fg = false;
+            }
+        }
+        if let Some(bg) = &self.bg {
+            bg_name = bg;
+            if theme.get(bg).bg.is_none() && theme.get(bg).add_modifier == Modifier::empty() {
+                found_bg = false;
+            }
+        }
+        if self.check_both {
+            if !found_fg {
+                messages.push(format!("{}.fg", fg_name.clone()));
+            }
+            if !found_bg {
+                messages.push(format!("{}.bg", bg_name.clone()));
+            }
+        } else {
+            if !found_fg && !found_bg {
+                messages.push(format!("{}", fg_name))
+            }
+        }
+    }
+}
+
+pub fn lint(file: String) -> Result<(), DynError> {
+    if file.contains("base16") {
+        println!("Skipping base16: {}", file);
+        return Ok(());
+    }
+    let path = path::themes().join(file.clone() + ".toml");
+    let theme = std::fs::read(&path).unwrap();
+    let theme: Theme = toml::from_slice(&theme).expect("Failed to parse theme");
+
+    let mut messages: Vec<String> = vec![];
+    get_rules()
+        .iter()
+        .for_each(|rule| rule.validate(&theme, &mut messages));
+
+    if messages.len() > 0 {
+        let message: String = messages
+            .iter()
+            .map(|m| {
+                let mut msg = file.clone();
+                msg.push_str(".");
+                msg.push_str(m);
+                msg
+            })
+            .collect::<Vec<String>>()
+            .join(" ")
+            .into();
+        println!("{}", message.replace(" ", "\n"));
+        Err(messages.len().to_string().into())
+    } else {
+        Ok(())
+    }
+}
+
+pub fn lint_all() -> Result<(), DynError> {
+    let files = std::fs::read_dir(path::themes())
+        .unwrap()
+        .filter_map(|entry| {
+            let path = entry.ok()?.path();
+            let name = path.file_name()?.to_string_lossy().to_string();
+            if path.is_dir() || name.contains("README") {
+                None
+            } else {
+                Some(name)
+            }
+        })
+        .collect::<Vec<String>>();
```

Comment:
Rather than writing these here, we can probably reuse helix-loader helper functions. As seen in `helix-view/src/theme.rs`
```rust
    pub fn read_names(path: &Path) -> Vec<String> {
        std::fs::read_dir(path)
            .map(|entries| {
                entries
                    .filter_map(|entry| {
                        let entry = entry.ok()?;
                        let path = entry.path();
                        (path.extension()? == "toml")
                            .then(|| path.file_stem().unwrap().to_string_lossy().into_owned())
                    })
                    .collect()
            })
            .unwrap_or_default()
    }
```